packages/permissions/src/utils/has-permissions.ts

```diff
@@ -16,6 +22,37 @@ type TActionRight = {
 type TActionRights = {
   [key: string]: TActionRight;
 };
+type TDataFence = {
```

Comment:
i'm no expert in types but the repetition is bugging me. can we have them in a central place and import them in these files ?

Reply:
As long as it's not a React component yes, otherwise the babel plugin to generate PropTypes doesn't work if types are in external files.

Reply:
so where does this leave me?
because this type is used in `authorized` (react) and `has-permissions` (not react)

Reply:
You can export the type from `authorized` and use it in the utility functions.

But we can also do that as a separate step...